graph/path/yen_ksp_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"math"
+	"reflect"
+	"testing"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+var yenShortestPathTests = []struct {
+	name  string
+	graph func() graph.WeightedEdgeAdder
+	edges []simple.WeightedEdge
+
+	query     simple.Edge
+	k         int
+	wantPaths [][]int64
+}{
+	// Positive weighted graphs.
+	// https://en.wikipedia.org/wiki/Yen%27s_algorithm
+	{
+		name:  "wikipedia example",
+		graph: func() graph.WeightedEdgeAdder { return simple.NewWeightedDirectedGraph(0, math.Inf(1)) },
+		edges: []simple.WeightedEdge{
+			{F: simple.Node(0), T: simple.Node(1), W: 3},
```

Comment:
Please make the edge and node definitions in the form e.g. `{F: simple.Node('C'), T: simple.Node('D'), W: 3},`,  `Query: simple.Edge{F: simple.Node('C'), T: simple.Node('H')},` and the wanted values the same, e.g. `{'C', 'E', 'F', 'H'},`.